pkg/scheduler/scheduler.go

```diff
@@ -230,10 +276,7 @@ func (k *KubernetesScheduler) InstallDebugHandlers(mux *http.ServeMux) {
 		func() {
```

Comment:
can actually eliminate this anonymous func() since it's only there to provide context for lock/unlock. instead of using a chan (get rid of that too), just set `slaves := k.slaves.getSlaveIds()` directly.